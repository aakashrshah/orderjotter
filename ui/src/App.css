:root {
  --main-bg-color: rgb(248, 249, 251);
}

body {
  background: var(--main-bg-color);
}

.content {
  padding: 25px 25px 25px 25px;
}

.future-div {
  height: 100%;
  width: 100%;
  background: lightblue;
  text-align: center;
}

.done-checkmark {
  color: rgb(0, 194, 33);
}
