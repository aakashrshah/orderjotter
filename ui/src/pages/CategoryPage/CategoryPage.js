import React from 'react';
import { connect } from 'react-redux';
import Grid from '@material-ui/core/Grid';

export function CategoryPage() {
  return (
    <Grid container spacing={1}>
      <Grid container spacing={1} direction="row" />
    </Grid>
  );
}

CategoryPage.propTypes = {
};

export default connect()(CategoryPage);
