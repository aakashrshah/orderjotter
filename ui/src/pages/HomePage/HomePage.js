/* eslint-disable no-restricted-imports */
import React from 'react';
import { connect } from 'react-redux';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import ExpansionPanel from '@material-ui/core/ExpansionPanel';
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import { makeStyles } from '@material-ui/core/styles';
import SpeedDial from '@material-ui/lab/SpeedDial';
import SpeedDialIcon from '@material-ui/lab/SpeedDialIcon';
import SpeedDialAction from '@material-ui/lab/SpeedDialAction';
import FileCopyIcon from '@material-ui/icons/FileCopyOutlined';
import SaveIcon from '@material-ui/icons/Save';
import PrintIcon from '@material-ui/icons/Print';
import ShareIcon from '@material-ui/icons/Share';
import FavoriteIcon from '@material-ui/icons/Favorite';
import NewOrderForm from '../../components/NewOrderForm/NewOrderForm';

const useStyles = makeStyles(theme => ({
  root: {
    width: '100%',
  },
  heading: {
    fontSize: theme.typography.pxToRem(15),
    fontWeight: theme.typography.fontWeightRegular,
  },
  speedDial: {
    position: 'absolute',
    '&.MuiSpeedDial-directionUp, &.MuiSpeedDial-directionLeft': {
      bottom: theme.spacing(2),
      right: theme.spacing(2),
    },
    '&.MuiSpeedDial-directionDown, &.MuiSpeedDial-directionRight': {
      top: theme.spacing(2),
      left: theme.spacing(2),
    },
  },
}));

const actions = [
  { icon: <FileCopyIcon />, name: 'Copy' },
  { icon: <SaveIcon />, name: 'Save' },
  { icon: <PrintIcon />, name: 'Print' },
  { icon: <ShareIcon />, name: 'Share' },
  { icon: <FavoriteIcon />, name: 'Like' },
];

export function HomePage() {
  const [expanded, setExpanded] = React.useState('panel1');
  const [open, setOpen] = React.useState(false);

  const handleClose = () => {
    setOpen(false);
  };

  const handleOpen = () => {
    setOpen(true);
  };
  const handleChange = panel => (event, newExpanded) => {
    setExpanded(newExpanded ? panel : false);
  };

  const classes = useStyles();

  return (
    <Grid container spacing={1}>
      <Grid container spacing={1} direction="row">
        <Grid item xs={12}>
          <Typography variant="h6">Welcome User</Typography>
        </Grid>
        <Grid container item xs={12} spacing={1}>
          <Grid item xs>
            <Grid container direction="row" justify="flex-start" alignItems="flex-start">
              <Grid item xs={12}>
                <div className={classes.root}>
                  <ExpansionPanel square expanded={expanded === 'panel1'} onChange={handleChange('panel1')}>
                    <ExpansionPanelSummary
                      expandIcon={<ExpandMoreIcon />}
                      aria-controls="panel1a-content"
                      id="panel1a-header"
                    >
                      <Typography className={classes.heading}><b>New order</b></Typography>
                    </ExpansionPanelSummary>
                    <ExpansionPanelDetails>
                      <NewOrderForm />
                    </ExpansionPanelDetails>
                  </ExpansionPanel>
                </div>
              </Grid>
            </Grid>
          </Grid>
        </Grid>
      </Grid>
      <div className={classes.exampleWrapper}>
        <SpeedDial
          ariaLabel="SpeedDial example"
          className={classes.speedDial}
          icon={<SpeedDialIcon />}
          onClose={handleClose}
          onOpen={handleOpen}
          open={open}
          direction="up"
        >
          {actions.map(action => (
            <SpeedDialAction
              key={action.name}
              icon={action.icon}
              tooltipTitle={action.name}
              onClick={handleClose}
            />
          ))}
        </SpeedDial>
      </div>
    </Grid>
  );
}

HomePage.propTypes = {
};

HomePage.defaultProps = {
};

function mapStateToProps(state) {
  console.log(state);
  return {
  };
}

export default connect(mapStateToProps)(HomePage);
