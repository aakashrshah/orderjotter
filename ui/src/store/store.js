/* global window */
/* eslint-disable no-underscore-dangle */
import { createStore, applyMiddleware, compose } from 'redux';
import thunkMiddleware from 'redux-thunk';

// eslint-disable-next-line no-restricted-imports
import rootReducer from '../reducers/reducers';

// NOTE: To enable redux-dev-tools support:
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  rootReducer,
  composeEnhancers(applyMiddleware(thunkMiddleware)),
);

export default store;
