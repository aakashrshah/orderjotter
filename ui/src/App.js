import React from 'react';
import { Route, Link } from 'react-router-dom';

import AppBar from '@material-ui/core/AppBar';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import Toolbar from '@material-ui/core/Toolbar';
import HomePage from './pages/HomePage/HomePage';
import LoginPage from './pages/LoginPage/LoginPage';
import CategoryPage from './pages/CategoryPage/CategoryPage';
import SignUpPage from './pages/SignUpPage/SignUpPage';
import PricingPage from './pages/PricingPage/PricingPage';
import DiscoverPage from './pages/DiscoverPage/DiscoverPage';
import ForgotPasswordPage from './pages/ForgotPasswordPage/ForgotPasswordPage';

import './App.css';

export default function App() {
  return (
    <React.Fragment>
      <AppBar position="static">
        <Toolbar>
          <Button color="inherit" component={Link} to="/">
            <Typography variant="h6" color="inherit" style={{ textTransform: 'none' }}>
              Order Jotter
            </Typography>
          </Button>
        </Toolbar>
      </AppBar>

      <div className="content">
        <Route exact path="/" component={LoginPage} />
        <Route exact path="/signup" component={SignUpPage} />
        <Route exact path="/home" component={HomePage} />
        <Route exact path="/interests" component={CategoryPage} />
        <Route exact path="/pricing" component={PricingPage} />
        <Route exact path="/explore" component={DiscoverPage} />
        <Route exact path="/forgotpassword" component={ForgotPasswordPage} />
      </div>
    </React.Fragment>
  );
}
