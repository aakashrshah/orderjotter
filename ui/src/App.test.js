/* globals beforeEach, describe, document, expect, it, jest */
import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { mount, shallow } from 'enzyme';
import { Link, MemoryRouter } from 'react-router-dom';

import Button from '@material-ui/core/Button';
import store from './store/store';
import App from './App';

import HomePage from './pages/HomePage/HomePage';
import CategoryPage from './pages/CategoryPage/CategoryPage';


describe('App', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <Provider store={store}>
          <App />
        </Provider>
      </MemoryRouter>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match a snapshot', () => {
    expect(shallow(<App />)).toMatchSnapshot();
  });

  it('should have an AppBar that routes to the home page.', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/project/ABC12345']}>
        <Provider store={store}>
          <App />
        </Provider>
      </MemoryRouter>,
    );
    const iqviaButton = wrapper.find(Button).at(0);
    expect(iqviaButton.prop('to')).toBe('/');
    expect(iqviaButton.prop('component')).toEqual(Link);
  });

  describe('Routing', () => {
    it('should route to the Category page', () => {
      const wrapper = mount(
        <MemoryRouter initialEntries={['/interests']}>
          <Provider store={store}>
            <App />
          </Provider>
        </MemoryRouter>,
      );
      expect(wrapper.find(CategoryPage)).toHaveLength(1);
    });
  });
});
