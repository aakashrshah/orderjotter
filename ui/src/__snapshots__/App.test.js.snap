// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match a snapshot 1`] = `
<Fragment>
  <WithStyles(ForwardRef(AppBar))
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(Button))
        color="inherit"
        component={[Function]}
        to="/"
      >
        <WithStyles(ForwardRef(Typography))
          color="inherit"
          style={
            Object {
              "textTransform": "none",
            }
          }
          variant="h6"
        >
          Order Jotter
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
  <div
    className="content"
  >
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/signup"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/home"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/interests"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/pricing"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/explore"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/forgotpassword"
    />
  </div>
</Fragment>
`;
